From 9ed73c823c987f606a4cd0f69a41f7485e1d1c81 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Mon, 4 Sep 2017 11:54:13 +0300
Subject: [PATCH 14/57] arm: imx7d: cl-som-imx7: add support for I2C4 bus

Add the following:
* Information structure for I2C 4.
* Initialization to I2C 4.
* CONFIG_SYS_I2C_MXC_I2C4 option to the board config.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 board/compulab/cl-som-imx7/cl-som-imx7.c | 24 ++++++++++++++++++++++++
 board/compulab/cl-som-imx7/common.h      |  3 +++
 include/configs/cl-som-imx7.h            |  1 +
 3 files changed, 28 insertions(+)

diff --git a/board/compulab/cl-som-imx7/cl-som-imx7.c b/board/compulab/cl-som-imx7/cl-som-imx7.c
index 05832087a3..28b706a961 100644
--- a/board/compulab/cl-som-imx7/cl-som-imx7.c
+++ b/board/compulab/cl-som-imx7/cl-som-imx7.c
@@ -29,6 +29,9 @@ static int nand_enabled = 0;
 
 #ifdef CONFIG_SYS_I2C_MXC
 
+/* Baseboard I2C bus is initialized flag */
+int cl_som_imx7_base_i2c_init;
+
 #define I2C_PAD_CTRL		(PAD_CTL_DSE_3P3V_32OHM | PAD_CTL_SRE_SLOW | \
 				PAD_CTL_HYS)
 
@@ -52,12 +55,33 @@ static struct i2c_pads_info cl_som_imx7_i2c_pad_info2 = {
 	},
 };
 
+static struct i2c_pads_info cl_som_imx7_i2c_pad_info4 = {
+	.scl = {
+		.i2c_mode = MX7D_PAD_GPIO1_IO10__I2C4_SCL |
+			MUX_PAD_CTRL(I2C_PAD_CTRL),
+		.gpio_mode = MX7D_PAD_GPIO1_IO10__GPIO1_IO10 |
+			MUX_PAD_CTRL(I2C_PAD_CTRL),
+		.gp = IMX_GPIO_NR(1, 10),
+	},
+	.sda = {
+		.i2c_mode = MX7D_PAD_GPIO1_IO11__I2C4_SDA |
+			MUX_PAD_CTRL(I2C_PAD_CTRL),
+		.gpio_mode = MX7D_PAD_GPIO1_IO11__GPIO1_IO11 |
+			MUX_PAD_CTRL(I2C_PAD_CTRL),
+		.gp = IMX_GPIO_NR(1, 11),
+	},
+};
+
 /*
  * cl_som_imx7_setup_i2c() - I2C  pinmux configuration.
  */
 static void cl_som_imx7_setup_i2c(void)
 {
+	int ret;
+
 	setup_i2c(0, CONFIG_SYS_I2C_SPEED, 0x7f, &cl_som_imx7_i2c_pad_info2);
+	ret = setup_i2c(1, CONFIG_SYS_I2C_SPEED, 0x7f, &cl_som_imx7_i2c_pad_info4);
+	cl_som_imx7_base_i2c_init = ret ? 0:1;
 }
 #else /* !CONFIG_SYS_I2C_MXC */
 static void cl_som_imx7_setup_i2c(void) {}
diff --git a/board/compulab/cl-som-imx7/common.h b/board/compulab/cl-som-imx7/common.h
index 3ed74015cd..c028701b3e 100644
--- a/board/compulab/cl-som-imx7/common.h
+++ b/board/compulab/cl-som-imx7/common.h
@@ -33,4 +33,7 @@ PADS_SET_PROT(wdog_pads)
 PADS_SET_PROT(gpmi_nand_pads)
 PADS_SET_PROT(nand_enable_pads)
 #endif /* CONFIG_NAND_MXS */
+#ifdef CONFIG_SYS_I2C_MXC
+extern int cl_som_imx7_base_i2c_init;
+#endif /* CONFIG_SYS_I2C_MXC */
 #endif /* !CONFIG_SPL_BUILD */
diff --git a/include/configs/cl-som-imx7.h b/include/configs/cl-som-imx7.h
index f4e009277a..195523261b 100644
--- a/include/configs/cl-som-imx7.h
+++ b/include/configs/cl-som-imx7.h
@@ -50,6 +50,7 @@
 #define CONFIG_SYS_I2C
 #define CONFIG_SYS_I2C_MXC
 #define CONFIG_SYS_I2C_MXC_I2C2		/* Enable I2C bus 2 */
+#define CONFIG_SYS_I2C_MXC_I2C4		/* Enable I2C bus 4 */
 #define CONFIG_SYS_I2C_SPEED		100000
 #define SYS_I2C_BUS_SOM			0
 
-- 
2.17.1

